junit-platform-commons/src/main/java/org/junit/platform/commons/util/ConsoleUtils.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2015-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+
+import java.nio.charset.Charset;
+
+import org.apiguardian.api.API;
+
+/**
+ * Collection of utilities for working with {@code java.io.Console}
+ * and friends.
+ *
+ * <h3>DISCLAIMER</h3>
+ *
+ * <p>These utilities are intended solely for usage within the JUnit framework
+ * itself. <strong>Any usage by external parties is not supported.</strong>
+ * Use at your own risk!
+ *
+ * @since 1.9
+ */
+@API(status = INTERNAL, since = "1.9")
+public class ConsoleUtils {
```

Comment:
I don't think this needs to be in junit-platform-commons but would move it to junit-platform-console and make it package-private.

Reply:
Thought of this too, but a having a source set that targets Java 17 directly in `junit-platform-commons` seems more future-proof to me, especially when it comes to overloading other common functionality.

Reply:
I don't see why we couldn't add it then, when we need it.

Reply:
Fine. Moving it to `junit-platform-console`.

Reply:
Moved via https://github.com/junit-team/junit5/pull/2882/commits/e187063e3de2891c31ed23931297f696c34d588d